### PR TITLE
PMM-6921 VMAgent changes for old versions

### PR DESCRIPTION
**Pull request by f41gh7**
now vmagents added, when client authenticates at pmm-server,
vmagent creation for pmm-server is disabled.

[PMM-6921](https://jira.percona.com/browse/PMM-6921)

- [x] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1245

- [ ] Links to other linked pull requests (optional).
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

**Review by BupycHuk — APPROVED** [review]

**Review by AlekSi — COMMENTED** [review]
There are merge conflicts now

File: services/agents/registry.go [hunk 1]
```diff
@@ -302,6 +313,58 @@ func authenticate(md *agentpb.AgentConnectMetadata, q *reform.Querier) (string,
...
+		}); err != nil {
+			return errors.Wrapf(err, "Can't create 'vmAgent' for pmm-agent with ID %q", pmmAgentID)
+		}
+	}
+	return nil
+}
+
+func removeVMAgentFromPMMAgent(q *reform.Querier, pmmAgentID string) error {
+	vmAgentType := models.VMAgentType
+	vmAgent, err := models.FindAgents(q, models.AgentFilters{PMMAgentID: pmmAgentID, AgentType: &vmAgentType})
```
Comment (BupycHuk) [inline services/agents/registry.go:341]: looks like it's better to do all these changes in one transaction.

Comment (BupycHuk) [reply services/agents/registry.go:341]: The same in `addVMAgentToPMMAgent`

Comment (AlekSi) [reply services/agents/registry.go:341]: Agree. `Registry.register` should use `r.db.InTransaction` and call `authenticate` with `tx`, not `r.db.Querier`

**Review by AlekSi — CHANGES_REQUESTED** [review]
https://github.com/percona/pmm-managed/pull/530#discussion_r530287910

File: services/agents/registry.go (hunk 1 above)
Comment (f41gh7) [reply services/agents/registry.go:354]: Thanks, added transaction.

**Review by BupycHuk — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Pull request merged**